Paraswap-Agents/AdminOperations/src/agent.spec.ts

```diff
@@ -0,0 +1,235 @@
+
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .setFrom(USER_ADDR)
+      // Add call to `setImplementation`
+      .addTraces({
+        to: IRRELEVENT_ADDR,
+        from: USER_ADDR,
+        input: SWAPPER_IFACE.encodeFunctionData(
+          "setImplementation",
+          TEST_DATA[3] // setImplementation args
+        ),
+      })
+      // Add call to `setFeeWallet`
+      .addTraces({
+        to: IRRELEVENT_ADDR,
+        from: USER_ADDR,
+        input: SWAPPER_IFACE.encodeFunctionData(
+          "setFeeWallet",
+          TEST_DATA[4] // setFeeWallet args
+        ),
+      })
+      // Add log of `RouterInitialized` event
+      .addAnonymousEventLog(IRRELEVENT_ADDR, log1.data, ...log1.topics);
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore other event logs and function calls on AugustusSwapper Contract", async () => {
+    const log1 = IRRELEVENT_IFACE.encodeEventLog(IRRELEVENT_IFACE.getEvent("wrongsig"), []);
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .setFrom(USER_ADDR)
+      // Add call to another function
+      .addTraces({
+        to: SWAPPER_ADDR,
+        from: USER_ADDR,
+        input: IRRELEVENT_IFACE.encodeFunctionData("wrongFunction", []),
+      })
+      .addAnonymousEventLog(SWAPPER_ADDR, log1.data, ...log1.topics);
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("Should return findings when admin operations are executed on AugustusSwapper Contract", async () => {
+    const log1 = SWAPPER_IFACE.encodeEventLog(SWAPPER_IFACE.getEvent("AdapterInitialized"), TEST_DATA[6]);
+
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .setFrom(USER_ADDR)
+      // Add call to `setImplementation`
+      .addTraces({
+        to: SWAPPER_ADDR,
+        from: USER_ADDR,
+        input: SWAPPER_IFACE.encodeFunctionData(
+          "setImplementation",
+          TEST_DATA[3] // setImplementation args
+        ),
+      })
+      // Add call to `setFeeWallet`
+      .addTraces({
+        to: SWAPPER_ADDR,
+        from: USER_ADDR,
+        input: SWAPPER_IFACE.encodeFunctionData(
+          "setFeeWallet",
+          TEST_DATA[4] // setFeeWallet args
+        ),
+      })
+      // Add call to `registerPartner`
+      .addTraces({
+        to: SWAPPER_ADDR,
+        from: USER_ADDR,
+        input: SWAPPER_IFACE.encodeFunctionData(
+          "registerPartner",
+          TEST_DATA[7] // setFeeWallet args
+        ),
+      })
+      // Add log of `AdapterInitialized` event
+      .addAnonymousEventLog(SWAPPER_ADDR, log1.data, ...log1.topics);
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([
+      createFinding("AdapterInitialized", TEST_DATA[6]),
+      createFinding("setImplementation", TEST_DATA[3]),
+      createFinding("setFeeWallet", TEST_DATA[4]),
+      createFinding("registerPartner", TEST_DATA[7]),
+    ]);
```

Comment:
Would be good to include the `RouterInitialized` event emission as well. All the others seem to be covered by the tests.